### ISSUE TITLE
losing precision in torch.sum

### ISSUE THREAD

**Issue by shivanimall** [pytorch/pytorch#117032]
### 🚀 The feature, motivation and pitch

Hello, I am losing precision when performing torch.sum operation, is it possible to avoid this? 

in case I am missing something, please let me know, thanks!

`(Pdb) numpy.sum(x_mem_banks_sim.cpu().numpy(), axis=0)
array([0.00019587, 0.00018269, 0.00018708, 0.00052095, 0.00020271,
       0.00019834, 0.00020154, 0.00018121], dtype=float32)
(Pdb) torch.sum(x_mem_banks_sim, dim=0, dtype=torch.float32)
tensor([0.0002, 0.0002, 0.0002, 0.0005, 0.0002, 0.0002, 0.0002, 0.0002],
       device='cuda:0')
(Pdb) torch.sum(x_mem_banks_sim, dim=0, dtype=torch.float64)
tensor([0.0002, 0.0002, 0.0002, 0.0005, 0.0002, 0.0002, 0.0002, 0.0002],
       device='cuda:0', dtype=torch.float64)
(Pdb) torch.sum(x_mem_banks_sim, dim=0, dtype=torch.double)
tensor([0.0002, 0.0002, 0.0002, 0.0005, 0.0002, 0.0002, 0.0002, 0.0002],
       device='cuda:0', dtype=torch.float64)

`

**Comment by malfet** [comment]
One would need to know how one created the data in the first place, but I wonder if `torch.set_printoptions(precision=8)` would make a difference (i.e. by default, when tensor is printed its rounded to 4 decimal points)
